airflow/ui/src/components/DataTable/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,188 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  FormControl,
+  FormLabel,
+  Input,
+  VStack,
+  ModalCloseButton,
+  Button,
+  ModalFooter,
+  Box,
+  Text,
+  Spacer,
+  HStack,
+  Collapse,
+} from "@chakra-ui/react";
+import { autocompletion } from "@codemirror/autocomplete";
+import { json } from "@codemirror/lang-json";
+import { oneDark } from "@codemirror/theme-one-dark";
+import CodeMirror, { lineNumbers } from "@uiw/react-codemirror";
+import React, { useState, useEffect } from "react";
+
+type DagParams = {
+  configJson: string;
+  dagId: string;
+  logicalDate: string;
+  runId?: string;
+};
+
+type TriggerDAGFormProps = {
+  dagParams: DagParams;
+  onClose: () => void;
+  onTrigger: () => void;
+  setDagParams: React.Dispatch<React.SetStateAction<DagParams>>;
+};
+
+const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
+  dagParams,
+  onTrigger,
+  setDagParams,
+}) => {
+  const [showDetails, setShowDetails] = useState(false);
+
+  useEffect(() => {
+    try {
+      const prettyJson = JSON.stringify(
+        JSON.parse(dagParams.configJson),
+        undefined,
+        2,
+      );
+
+      setDagParams((prev) => ({ ...prev, configJson: prettyJson }));
+    } catch {
+      // Invalid JSON handling
+    }
+  }, [dagParams.configJson, setDagParams]);
+
+  const handleChange = (
+    ele: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = ele.target;
+
+    setDagParams((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleReset = () => {
+    setDagParams({
+      configJson: "{}",
+      dagId: dagParams.dagId,
+      logicalDate: "",
+      runId: "",
+    });
+  };
+
+  const handleJsonChange = (value: string) => {
+    setDagParams((prev) => ({ ...prev, configJson: value }));
+  };
+
+  const isValidJson = () => {
+    try {
+      JSON.parse(dagParams.configJson);
+
+      return true;
+    } catch {
+      return false;
+    }
+  };
+
+  return (
+    <>
+      <ModalCloseButton />
+
+      <VStack align="stretch" p={5} spacing={2}>
+        <Button
+          onClick={() => setShowDetails(!showDetails)}
+          variant="outline"
+          width="full"
+        >
+          {showDetails ? "Hide Advance Options" : "Show Advance Options"}
+        </Button>
+
+        <Collapse in={showDetails}>
+          <VStack align="stretch" spacing={3}>
+            <FormControl>
+              <FormLabel size="sm">Logical date</FormLabel>
+              <Input
```

Comment:
I am not sure and am not really an expert but @bbovenzi and @pierrejeambrun proposed to me to use react-hook-form as toolkit for the trigger form. Don't know if this is proposed instead of a plain form.